src/bvh/bvh.rs

```diff
@@ -200,6 +200,14 @@ impl BVHNode {
         }
     }
 
+    /// Returns the index of the shape contained in the node, assuming it is a leaf.
+    pub fn shape_index(&self) -> usize {
+        match *self {
+            BVHNode::Leaf { shape_index, .. } => shape_index,
+            _ => panic!("Tried to get the shape index of an interior node."),
```

Comment:
I'm not sure this should panic. Can't we rather define this to be `None` and return and `Option`?

Reply:
Yeah, that does sound better. One sec!

Reply:
`panic!` should happen if you get run an invalid operation and I can't quite decide whether this is an invalid operation or whether it's actually valid to run this on non-`Leaf` nodes. What do you think?

Reply:
I think Option is a bit prettier. I wasn't too fond of having to deconstruct the enum just to test whether it's a node or a leaf, *and then* having the `shape_index` fn do another runtime check of self's variant.